Source/VungleAdapterBannerAd.swift

```diff
@@ -5,268 +5,146 @@
-        adWasDisplayed = true
-    }
-    
-    private func finishDisplayingVungleAd() {
-        // Programmatic
-        if let adm = request.adm {
-            VungleSDK.shared().finishDisplayingAd(request.partnerPlacement, adMarkup: adm)
-        } else {
-        // Non-programmatic
-            VungleSDK.shared().finishDisplayingAd(request.partnerPlacement)
-        }
     }
-    
+
     /// Mapping of the request ad size to Vungle's ad size type. Nil means a MREC banner.
-    private var vungleAdSize: VungleAdSize? {
-        let size = request.size ?? IABStandardAdSize
-        switch size {
-        case IABMediumAdSize:
-            // Note that Vungle MREC ads are not considered banners by Vungle and require using a different set of methods.
-            // We return nil here to signify that this is not a Vungle banner, but an MREC ad.
-            return nil
+    private var vungleAdSize: BannerSize {
+        switch request.size {
+        case IABStandardAdSize:
+            return BannerSize.regular
+        case CGSize(width: 300, height: 50):
+            return BannerSize.short
         case IABLeaderboardAdSize:
-            return .bannerLeaderboard
+            return BannerSize.leaderboard
+        case IABMediumAdSize:
+            return BannerSize.mrec
         default:
-            return .banner
-        }
-    }
-    
-    /// A size for the Vungle banner/MREC ad container compatible with that specific Vungle ad type.
-    /// Note that this size must match Vungle's expectations or the ad rendering will fail.
-    /// See Vungle's documentation on banner ads: https://support.vungle.com/hc/en-us/articles/360048572771
-    /// See Vungle's documentation on MREC ads: https://support.vungle.com/hc/en-us/articles/360048079292
-    private var adContainerSize: CGSize {
-        switch vungleAdSize {
-        case nil:
-            // A nil vungleAdSize means the ad is not technically a Vungle banner, but a Vungle MREC ad.
-            // MREC ads must always have a size 300x250 per Vungle's documentation.
-            return IABMediumAdSize
-        case .banner:
-            return IABStandardAdSize
-        case .bannerShort:
-            return CGSize(width: 300, height: 50)
-        case .bannerLeaderboard:
-            return IABLeaderboardAdSize
-        case .unknown:
-            return IABStandardAdSize
-        @unknown default:
-            return IABStandardAdSize
+            log("Unrecognized banner size.")
+            // Ad is unlikely to load if the size doesn't match, but we need to return something
+            return BannerSize.regular
         }
     }
 }
 
-// MARK: - VungleSDKHBDelegate
+// MARK: - VungleBannerDelegate
+extension VungleAdapterBannerAd: VungleBannerDelegate {
+    func bannerAdDidLoad(_ banner: VungleBanner) {
+        log(.loadSucceeded)
+        loadCompletion?(.success([:])) ?? log(.loadResultIgnored)
```

Comment:
Should we fail the load if any of the subsequent checks (ad is nil, ad can't play, etc.) fail?
Otherwise we will end up with a blank space for 30 seconds. If Vungle can't show the add then effectively they have failed the load.

Reply:
[Done](https://github.com/ChartBoost/chartboost-mediation-ios-adapter-vungle/pull/44/commits/247179ce1f143b9d9d959384c022a4b3dc41f187), because we have to avoid that blank 30 seconds but it made more sense to me the way it was - if the partner says it loaded, call that success and anything after that is a show error. Why doesn't a failed `show()` immediately trigger a new call to `load()`?

Reply:
The issue is that we don't have a concept of banner show errors. A loaded helium banner is supposed to be able to be displayed. If it's not, it's not really loaded (according to helium's internal definition).
If later the partner reports a show failure that should probably be mapped to a didDismiss callback. Although at the moment Helium will just ignore dismiss calls for banners, at some point we could handle that better and trigger a reload.